packages/jest-serializer/src/index.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright (c) 2018-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+'use strict';
+
+import fs from 'fs';
+import v8 from 'v8';
+
+import type {Path} from 'types/Config';
+
+// JSON and V8 serializers are both stable when it comes to compatibility. The
+// current JSON specification is well defined in RFC 8259, and V8 ensures that
+// the versions are compatible by encoding the serialization version in the own
+// generated buffer.
+
+const JS_TYPE = '__$t__';
+const JS_VALUE = '__$v__';
+const JS_VF = '__$f__';
+
+function replacer(key: string, value: any): any {
+  // NaN cannot be in a switch statement, because NaN !== NaN.
+  if (Number.isNaN(value)) {
+    return {[JS_TYPE]: 'n'};
+  }
+
+  switch (value) {
+    case undefined:
+      return {[JS_TYPE]: 'u'};
+
+    case +Infinity:
+      return {[JS_TYPE]: '+'};
+
+    case -Infinity:
+      return {[JS_TYPE]: '-'};
+  }
+
+  switch (value && value.constructor) {
+    case Date:
+      return {[JS_TYPE]: 'd', [JS_VALUE]: value.getTime()};
+
+    case RegExp:
+      return {[JS_TYPE]: 'r', [JS_VALUE]: value.source, [JS_VF]: value.flags};
+
+    case Set:
+      return {[JS_TYPE]: 's', [JS_VALUE]: Array.from(value)};
+
+    case Map:
+      return {[JS_TYPE]: 'm', [JS_VALUE]: Array.from(value)};
+
+    case Buffer:
+      return {[JS_TYPE]: 'b', [JS_VALUE]: value.toString('latin1')};
```

Comment:
Any particular reason behind `latin1` encoding?

Reply:
It's the most optimal way of converting a buffer into a string. Other candidates would be an array (way too long to decode), a UTF-8 string (but not all sequences you might find are valid sequences), or a hex one (which is going to longer or at most as equal as a latin1 string) 🙂